src/frontend/src/optimizer/plan_node/stream_source.rs

```diff
@@ -75,7 +75,7 @@ impl StreamNode for StreamSource {
             source_id: source_catalog.id,
             source_name: source_catalog.name.clone(),
             state_table: Some(
-                generic::Source::infer_internal_table_catalog()
+                generic::Source::infer_internal_table_catalog(false)
```

Comment:
May add doc on why the dist key is not required for the stream source case.

Reply:
done